pkg/addons/default/kube_proxy.go

```diff
@@ -97,6 +110,38 @@ func UpdateKubeProxyImageTag(clientSet kubernetes.Interface, controlPlaneVersion
 	return false, nil
 }
 
+func isMissingArm64NodeSelector(daemonSet *v1.DaemonSet) bool {
+	if daemonSet.Spec.Template.Spec.Affinity != nil &&
+		daemonSet.Spec.Template.Spec.Affinity.NodeAffinity != nil &&
+		daemonSet.Spec.Template.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution != nil {
```

Comment:
:vomiting_face: I think this is unavoidable AFAIK

Reply:
😭